doc/source/skips/1-governance.rst

```diff
@@ -0,0 +1,165 @@
+pull requests, can cast votes for and against merging a pull-request, and can
+be involved in deciding major changes to the API. Core developers are expected
+to adhere to the :ref:`core developer guide <core_dev>`.
+
+New core developers can be nominated by any existing core developer.
+Discussion about new core developer nominations is one of the few activities
+that takes place on the project's private management list. The decision to
+invite a new core developer must be made by “lazy consensus”, meaning unanimous
+agreement by all responding existing core developers. Invitation must take
+place at least one week after initial nomination, to allow existing members
+time to voice any objections.
+
+Steering Council
+----------------
+The Steering Council (SC) members are core developers who have additional
+responsibilities to ensure the smooth running of the project. SC members are
+expected to participate in strategic planning, approve changes to the
+governance model, and make decisions about funding granted to the project
+itself. (Funding to community members is theirs to pursue and
+manage.) The purpose of the SC is to ensure smooth progress from the
+big-picture perspective. Changes that impact the full project require a
+synthetic analysis and a consensus that is both explicit and informed. When the
+core developer community (including the SC members) fails to reach such a
+consensus in a reasonable timeframe, the SC is the entity to resolve the issue.
+
+Membership of the SC is by nomination by a core developer. A nomination will
+result in discussion that cannot take more than a month and then admission to
+the SC by consensus. SC members who do not actively engage
+with the SC duties are expected to resign.
+
+The initial Steering Council of scikit-image consists of Stéfan van der
+Walt, Juan Nunez-Iglesias, Emmanuelle Gouillart, and Josh Warner.
+
+Decision Making Process
+=======================
+Decisions about the future of the project are made through discussion with all
+members of the community. All non-sensitive project management discussion takes
+place on the project `mailing list <mailto:scikit-image@python.org>`_
+and the `issue tracker <https://github.com/scikit-image/scikit-image/issues>`_.
+Occasionally, sensitive discussion may occur on a private list.
+
+Decisions should be made in accordance with the :ref:`mission, vision and
+values <values>` of the scikit-image project.
+
+Scikit-image uses a “consensus seeking” process for making decisions. The group
+tries to find a resolution that has no open objections among core developers.
+If no option can be found without objections, the decision is escalated to the
+SC, which will itself use consensus seeking to come to a resolution. In the
+unlikely event that there is still a deadlock, the proposal will move forward
+if it has the support of a simple majority of the SC. Any vote must be
+backed by a :ref:`scikit-image proposal (SKIP) <skip>`.
```

Comment:
Perhaps this is too strict a requirement?  I can be convinced with a good motivation.

Reply:
I think this is a good requirement specifically *for a vote*. Any time that consensus can be reached (quickly), a SKIP is probably not needed. Basically the proposed view is:

- propose a change. If everyone agrees, go for it. Otherwise,
- try to convince any dissenters via the normal routes (on GitHub threads, mostly), if that works (possibly with some modifications), go for it. Otherwise,
- try to make your case with a SKIP.

In short, majority votes *never* decide a PR's merge status without a SKIP.

If you agree with this, I'll rephrase to make that workflow more clear.

Reply:
> In short, majority votes never decide a PR's merge status without a SKIP.

I'd prefer to rephrase this as a (strong) recommendation but not a hard requirement.

I can easily imagine a point of conflict that is largely based on subjective opinion/experience were a SKIP wouldn't be very useful but a vote would be the natural tool to resolve a "deadlock". If a conflict requires a vote but doesn't generate a significant discussion I'd argue that a SKIP is wasted work.

Maybe I misunderstand the purpose and scope of SKIPs but I feel that SKIPs are most useful to structure a diverse discussion with many points that can or have to be considered. It addresses the shortcomings of discussions on classical channels like issues, PRs and the mailing list. Those forums are largely chronological and tend to bury older talking points and often drift off topic while a SKIP collects and presents the ongoing state of a discussion in a structured way.

Reply:
*If* consensus cannot be reached within a GitHub discussion, then *by definition* a deep argument is needed, and that is the purpose of the SKIP. The basic premise of the governance doc (and something that @stefanv has pushed for very strongly above) is that voting should almost never happen.

Core developers should know the distinction between "I can grudgingly live with this" and "this change is fundamentally flawed and I don't think it should be in the library". If a majority of developers are happy with a change and one holds the former opinion, then the change can go through. If even one core developer holds the latter view, then imho a SKIP is definitely required!

Reply:
I see both of your points, and while I was leaning towards @lagru's perspective yesterday when I wrote this, I think @jni is right in that we should almost never hit this point in the process, and when we do writing it up probably makes sense.  If people are bikeshedding something trivial and can't agree then they should go back and get better at finding consensus.  It is exactly right that consensus does not mean that we necessarily do what we all think is best, just what we all think is a reasonable path forward that we can live with.

Reply:
> Core developers should know the distinction between "I can grudgingly live with this" and "this change is fundamentally flawed and I don't think it should be in the library". If a majority of developers are happy with a change and one holds the former opinion, then the change can go through. If even one core developer holds the latter view, then imho a SKIP is definitely required!

I hadn't considered this. I guess no core developer would force a vote for trivial reasons or a trivial matter and in that case that's a good premise for a SKIP. Thanks for the insight, I'm convinced. :)

Reply:
I think a SKIP is fine when code is involved, but an SC vote may be used to determine some other decision, like whether to accept funding from a specific source, etc.  In that case, a SKIP may not make sense.